src/Debugging/Debugger.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Debugging;
+
+use Saloon\Contracts\DebuggingDriver;
+use Saloon\Debugging\Drivers\RayDebugger;
+use Saloon\Exceptions\UnknownDriverException;
+use Saloon\Debugging\Drivers\ErrorLogDebugger;
+use Saloon\Debugging\Drivers\SystemLogDebugger;
+
+class Debugger
+{
+    /**
+     * Globally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected static array $globalRegisteredDrivers = [];
+
+    /**
+     * Locally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected array $registeredDrivers = [];
+
+    /**
+     * Drivers that have been subscribed to
+     *
+     * @var array<int, string>
+     */
+    protected array $useDrivers = [];
+
+    /**
+     * Denotes if we send the request to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
```

Comment:
I think it might be ideal to somehow very if the driver-requirements are installed.
syslog and error log shouldn't be an issue, but I think we should somehow check if Ray is installed, before registering its driver.

Maybe I'm overthinking it, though.
People would still need to actively choose which driver(s) to use, even if they are registered.
Maybe we should add a new method to the interface like.. `hasPreRequisites()`, or something, in which we check prerequisites like dependencies, and return true/false.
Then in `useDriver()`, we can check if it has the prerequisites, and, if not, throw an Exception that tells the user which prerequisites it needs.

Reply:
Ooooh I really like that! `hasDependencies`?

Reply:
So what would you imagine we do, should we have a check while registering that checks if "canBeRegistered" or something like that (hasPreRequisites) and then if it doesn't it could just not register that driver? Or we could throw an exception. 

Come to think of it, it seems like quite a big of extra logic. I do agree with you though that having ray registered by default seems a bit odd. Oh I also removed the stream debugger just as I didn't feel it had a place, I've never thought to stream to a file.

Reply:
I think it's better at the `useDriver()` stage.
The reason is because it's only ever there that the user takes an active choice which driver(s) to use for debugging.